python/ray/train/_internal/storage.py

```diff
@@ -451,6 +452,10 @@ def __init__(
         self._create_validation_file()
         self._check_validation_file()
 
+        # Timestamp is used to create a unique session directory for the current
+        # training job. This is used to avoid conflicts when multiple training jobs
+        # run with the same name in the same cluster.
+        # This is set ONCE at the creation of the storage context, on the driver.
```

Comment:
Can you help me remember how did we resolved the consistency issue with the timestamp. Did we resolve it by writing files into driver directory and using background syncing.

Reply:
See the first bullet point in the PR description under "Some undesirable workarounds made in this PR."